pkg/rslog/output_builder_test.go

```diff
@@ -89,11 +123,20 @@ func (s *LoggerSuite) TestBuildOpenFileError() {
 		"/default/log/logfile.log",
 		os.O_APPEND|os.O_CREATE|os.O_WRONLY,
 		os.FileMode(0600),
-	).Return(defaultLogPath, nil)
+	).Return(expectedWriter, nil)
+
+	result, err := builder.Build(
+		OutputDest{
+			Output:   LogOutputFile,
+			Filepath: "/custom/test.log",
+		})
 
-	result, err := builder.Build(LogOutputFile, "/custom/test.log")
 	s.Nil(err)
-	s.Equal(result, defaultLogPath)
```

Comment:
Yeah, the thing is, now the builder returns an `io.multiWriter`. But, unfortunately, this struct is not public on the `io` package, I can't check if the result is that, and I can't do a contains this `writerMock` on that slice of `writers`, for example. 

So the solution to that:
- As it is a struct, calling the `Write` function, it will call the `Write` of all the writers inside it; 
- I'm filling it with mock writers and checking If the writer I want is receiving a call to the `Writer` function when I call this function of the `multiWriter` returned.

About your second question, what makes a difference here is the different options and mocked stuff on the `mockFileSystem.` If the parameters passed on the mocking statement don't match the ones in the testing phase, it will make the test fail.